### ISSUE TITLE
Evaluate if we can remove `do_action( 'woocommerce_sidebar' );` from Legacy Template block

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/woocommerce-gutenberg-products-block#5077]
A couple of templates rendered by the Legacy Template block [run](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/src/BlockTypes/LegacyTemplate.php#L88) `do_action( 'woocommerce_sidebar' );`, that causes an error to appear in debug mode:

![imatge](https://user-images.githubusercontent.com/3616980/140504876-fc18a88a-55ce-4fbd-bc33-a279253b55b4.png)

and renders an unstyled sidebar below the block given that the styles are not loaded.

This issue is about investigating what are the pros and cons of removing the call to that action.

**Comment by Aljullu** [comment]
Another option is to keep the `woocommerce_sidebar` action but unregister the `woocommerce_get_sidebar` function [associated with it](https://github.com/woocommerce/woocommerce/blob/3611d4643791bad87a0d3e6e73e031bb80447417/plugins/woocommerce/includes/wc-template-hooks.php#L54). This is what WC does for the WP core themes without sidebar:

https://github.com/woocommerce/woocommerce/blob/3611d4643791bad87a0d3e6e73e031bb80447417/plugins/woocommerce/includes/theme-support/class-wc-twenty-twenty-one.php#L28

However, IMO we should remove the `do_action( 'woocommerce_sidebar' );` call altogether from the Legacy Template block. Block templates won't be rendering the WooCommerce sidebar, so I don't think it makes sense calling that action at all. In addition, block themes seem like a good opportunity to not carry over this action. I took a quick (and non-exhaustive) look at how it's being used by extensions and all instances I found are about [hiding](https://github.com/woocommerce/retired-themes/blob/5922fb6038b98048ca4732175ea619bb06915990/briefed-commerce/woocommerce/woocommerce-layout.php) and [replacing the sidebar](https://github.com/woocommerce/spectrum/blob/d43e61352ab310a31d939a82adedd414414c7fde/includes/integrations/woocommerce/setup.php#L59-L60). But those things can now be done via FSE technologies.

So my proposal would be to remove `do_action( 'woocommerce_sidebar' );` from all block templates rendered by the Legacy Template block. In the unlikely case that we get reports that it broke extensions, we can reconsider adding it back (maybe with `remove_action( 'woocommerce_sidebar', 'woocommerce_get_sidebar', 10 );`), but I would only do that if really necessary. In fact, I expect the possibility to break extensions to be higher if we keep the action than if we remove it.

Thoughts @frontdevde @tjcafferkey?
